pom.xml

```diff
@@ -108,6 +108,15 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.6.0</version>
+				<configuration>
+					<source>1.8</source>
+					<target>1.8</target>
```

Comment:
I was actually thinking that our baseline should be 1.8 for all projects and just create a 1.5 compiled version of it for projects that uses older version. The reason I think is as s a Java framework, we should always encourage developers to use the latest and greatest Java stable/mainstream version. It's not that we're mandating them, they can still modify the source target to use 1.5 - but IMO, they might missed out on the new features post JDK 1.5 (GC1 as oppose to parallel/cms, Lambda, FileIO etc).

Also, I can imagine that majority (maybe 90% or more) of the chatbot developers would create their solution from scratch. ChatBot will have a lot of performance issues if it's coupled with an existing project since it can hog up resources - depending on the number of responses. It wouldn't be wise to design a chatbot solution to be incorporated in a project - it should stand alone on it's own, living in it's own virtual machine, using it's own allocated space - think microservices / soa pattern approach.

To reiterate though, we're not mandating people to use 1.8 - but I'd like our technology to use the latest and publicly/generally accepted version as our base. 

You're call on this one. Mine is merely a suggestion. I can remove the compiler plugin and remove the multiple exception catch (only available in 1.7 and higher jdk) code, No big deal. :)